CMakeLists.txt

```diff
@@ -203,17 +230,18 @@ include(GNUInstallDirs)
 
 option(BUILD_SHARED_LIBS "Build shared library" ON)
 if(BUILD_SHARED_LIBS)
+    set(CMAKE_VERBOSE_MAKEFILE ON)
     add_library(realsense SHARED ${REALSENSE_CPP} ${REALSENSE_HPP} ${REALSENSE_DEF})
     set_target_properties(realsense PROPERTIES VERSION ${REALSENSE_VERSION_STRING}
                                     SOVERSION ${REALSENSE_VERSION_MAJOR})
-    target_link_libraries(realsense ${LIBUSB1_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+    target_link_libraries(realsense ${LIBUSB1_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${gpu_kernel_library} ${CUDA_LIBRARIES} ${CATKIN_LIBRARIES} )
 else()
-    add_library(realsense STATIC ${REALSENSE_CPP} ${REALSENSE_HPP})
+    add_library(realsense STATIC ${REALSENSE_CPP} ${REALSENSE_HPP} ${gpu_kernel_library} ${CUDA_LIBRARIES} ${CATKIN_LIBRARIES})
 endif()
 
 target_include_directories(realsense PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-                                            $<INSTALL_INTERFACE:include>
-                                     PRIVATE ${LIBUSB1_INCLUDE_DIRS})
+                                            $<INSTALL_INTERFACE:include> /opt/ros/kinetic/include
```

Comment:
I don't think librealsense should depend on ROS.

Reply:
i used this for debugging and logging. i will remove it before final merge.